crates/sargon/src/wrapped_radix_engine_toolkit/high_level/manifest_building/assert_manifest.rs

```diff
@@ -53,3 +53,33 @@ pub fn instructions_eq(instructions: Instructions, expected: impl AsRef<str>) {
         trim(expected.as_ref())
     );
 }
+
+/// This is a **lenient** manifest InstructionV2 string compare asserting method, meaning it is
+/// possible it might regard two in fact different InstructionsV2 set as equal since it
+/// performs some string replacing which are breaking the actual Instructions set in the
+/// sense that it would result in InstructionV2 String roundtrip failure, i.e.
+/// parsing the string we change to might result in parsing error. It will also
+/// make these two different InstructionsV2 set equal each other:
+/// "Foo bar"
+/// "Foobar"
+/// where that string might be the value of some metadata key, e.g. a manifest
+/// creating a fungible token.
+/// However, the purpose of this assert is to simplify implementation of tests
+/// of InstructionsV2 set use to capture unintentional changes to the instructions of the
+/// manifest, i.e. which instructions that go in it, and the order of these.
+/// It is quite tedious to exactly match a Instructions set
+/// string without this test method since every space need to be correct.
+/// This function also allows for prettier formatting of the InstructionsV2 set string
+/// we assert against, since we can use any number of tabs.
+#[cfg(test)]
+pub fn instructions_v2_eq(
+    instructions: InstructionsV2,
+    expected: impl AsRef<str>,
+) {
+    let trim =
+        |s: &str| s.replace(' ', "").replace('\t', " ").trim().to_owned();
+    pretty_assertions::assert_eq!(
+        trim(&instructions.to_string()),
+        trim(expected.as_ref())
+    );
```

Comment:
You could create a common function between the two instructions versions, after all the `trim` and assert_eq works on strings.